agent/consul/acl.go

```diff
@@ -1132,6 +1133,27 @@ func (a ACLResolveResult) AccessorID() string {
 	return a.ACLIdentity.ID()
 }
 
+type ACLKVResourceID interface {
+	KVResourceID() string
```

Comment:
I'm not sure if we want a separate interface for each resource type, or a small number of interfaces (one for resources with identifiers+entMeta, one for just identifiers, and one for just entMeta).